src/data/navigation.ts

```diff
@@ -112,6 +112,21 @@ export const navigation: NavigationItem[] = [
           {
             title: "Search Examples", href: "/code-search/examples",
           },
+          {
+            title: "Code Navigation", href: "/code-search/code-navigation",
+            subsections: [
+              { title: "Capabilities", href: "/code-search/code-navigation/features", },
+              { title: "Search Navigation", href: "/code-search/code-navigation/search_based_code_navigation", },
+              { title: "Precise Navigation", href: "/code-search/code-navigation/precise_code_navigation", },
+              { title: "Indexers", href: "/code-search/code-navigation/writing_an_indexer", },
+              { title: "Auto-indexing", href: "/code-search/code-navigation/auto_indexing", },
+              { title: "Environment Variables", href: "/code-search/code-navigation/envvars", },
+              { title: "Troubleshooting", href: "/code-search/code-navigation/troubleshooting", },
```

Comment:
One of the things that we were struggling with - should there be a top-level Troubleshooting doc? Because internally, it needs to mimic the higher level structure. For example, we will need different steps for trouble-shooting search-based code nav, precise code nav and auto-indexing.

So should all of that live under Troubleshooting and just be linked from the corresponding section? Or should it all live under the appropriate section and be linked from Troubleshooting? Is there a reason to prefer one over the other?